src/vsr/replica.zig

```diff
@@ -6659,19 +6691,19 @@ const DVCQuorum = struct {
 
         var commit_max_: u64 = 0;
         for (dvcs.constSlice()) |dvc| {
-            const dvc_headers = message_body_as_headers_chain_disjoint(dvc);
-            // The DVC is connected to the committed ops.
-            const dvc_commit_max_connected = dvc_headers[dvc_headers.len - 1].op -| 1;
+            const dvc_headers = message_body_as_view_headers(dvc);
+            // DVC generation stops when a header with op ≤ commit_max is appended.
+            const dvc_commit_max_tail = dvc_headers.slice[dvc_headers.slice.len - 1].op;
```

Comment:
Can `dvc_headers.slice[dvc_headers.slice.len - 1]` be invalid?

Reply:
Invalid in what sense?

`dvc_headers.slice.len` is always at least 1.
`dvc_headers.slice[dvc_headers.slice.len - 1]` may correspond to a blank or a fault, but in those cases we would still know that their op (which is set even for blanks/faults) corresponds to `commit_max` according to the sender.

Reply:
blank or fault, but we indeed set `op` correctly for them